### ISSUE TITLE
For Analytics - errors should provide users with a means to debug failures

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#768]
See attached screenshot - the cause of the failure was unknown - but the user had no recourse other than to recreate the project as there is no feature in the UI that provides more detailed error/failure reporting.

![screenshot 24 _li](https://user-images.githubusercontent.com/642621/30130482-54692b56-9317-11e7-9312-5419cb90a007.jpg)

**Comment by joshuawilson** [comment]
Was this fixed?

**Comment by ldimaggi** [comment]
No - this is still an issue.

**Comment by sivaavkd** [comment]
@samuzzal-choudhury - can you please review this ?

**Comment by samuzzal-choudhury** [comment]
@ldimaggi this error occurs on receiving HTTP error 500 from the server. The most obvious remedy in such a scenario will be to retry the pipeline run so that stack analysis on recommender API is re-invoked.

Could you please help me understand what kind of a user experience are you referring to for such a scenario?

**Comment by muruGanesan** [comment]
If there is a HTTP server error,  we need to display an 'Error page' with details and provide option to the user to re-do/redirect.
From UX side, I will provide common server error page design - that should be displayed whenever user come across such error.

**Comment by samuzzal-choudhury** [comment]
@muruGanesan as you already know, Stack Report Recommendation is a part of the space dashboard. Do you intend to provide an error page for that section only?

**Comment by mindreeper2420** [comment]
UX already has a set of error pages, based on received errors (401,500, etc). They should be in InVision.

This is also just a card error, and should be contained to just the card. As there are already `if` statements for blank card vs one with content, another should be added for any kind of error. 

I’ll create an issue in the UX backlog and link to this.

**Comment by mindreeper2420** [comment]
I’ve created a UX issue here for this problem: https://github.com/fabric8-ui/fabric8-ux/issues/752. We triage and plan on Mondays.

**Comment by samuzzal-choudhury** [comment]
This issue cannot be targeted for analytics sprint planning unless the UX team take action on the related UX issue #752. Hence, moving #768 to "Analytics Backlog".

**Comment by kbsingh** [comment]
@samuzzal-choudhury are you sure about #752 ? that looks entirely unrelated.

**Comment by samuzzal-choudhury** [comment]
@kbsingh thanks for pointing that out. It should have been https://github.com/fabric8-ui/fabric8-ux/issues/752. I updated that.

**Comment by muruGanesan** [comment]
@samuzzal-choudhury @kbsingh , @mindreeper2420,  @sivaavkd @invincibleJai 

https://redhat.invisionapp.com/share/KGE3V2B79#/265822086_OSIO_Analytics_V07_Error-001

**Comment by invincibleJai** [comment]
@muruGanesan thanks for help

**Comment by jyasveer** [comment]
@muruGanesan I assume the button in the new UX page for the analytics dashboard is to take a user to the `Create pipeline` screen. But the text in the button says `Restart from pipe-line`. Can we change this to a appropriate text please ?

**Comment by muruGanesan** [comment]
@jyasveer , @samuzzal-choudhury , @invincibleJai 
I updated the screen please look at:
https://redhat.invisionapp.com/share/KGE3V2B79#/screens/265852226

**Comment by ldimaggi** [comment]
Will we also include error details, in addition to the pipeline restart button?

**Comment by Preeticp** [comment]
I have added comments in the Invision doc, just to add to that:
Afaik, to restart a pipeline the user would need to go to the Pipelines page, click the Kebab Menu and then **Start Pipeline**.
"Restart from Pipeline" does not convey this.

It would be great, if the button there actually restarts the Pipeline as **Pipelines>Start Pipeline** does. That would be ideal user experience and we could simply call it "Restart Pipeline". Can we implement that?

Currently clicking the button simply takes the user to the Pipelines screen, so instead of the button, can we simply have a line under 'We encountered an unexpected server error' saying 'Switch to the Pipelines<link to Pipelines screen> view to restart the pipeline.'
Wdyt?
